src/QueryClause.ts

```diff
@@ -0,0 +1,334 @@
+/**
+ * Base QueryClause implementation.
+ */
+export abstract class QueryClause {
+    abstract toJson(): any;
```

Comment:
Sorry for lack of my knowledge but I want to ask you one more.

Does not typescript has access modifier like `internal` in swift or `package internal` in java?